showyourwork/workflow/scripts/pdf.py

```diff
@@ -23,9 +23,22 @@
     \OnGithubActionsfalse
     ((* endif *))
     \def\syw@url{((- git_url -))}
+    \def\syw@shortUrl{((- git_short_url -))}
+    \def\syw@slug{((- git_slug -))}
     \def\syw@sha{((- git_sha -))}
     \def\syw@runid{((- github_runid -))}
 
+    ((* if stamp.enabled *))
+    \AddStamptrue
+    ((* else *))
+    \AddStampfalse
+    ((* endif *))
+    \newcommand{\syw@stampX}{((- stamp.xpos -))}
+    \newcommand{\syw@stampY}{((- stamp.ypos -))}
+    \newcommand{\syw@stampSize}{((- stamp.size -))}
+    \newcommand{\syw@stampAngle}{((- stamp.angle -))}
+    \newcommand{\syw@stampFontSize}{\footnotesize}
```

Comment:
My thought was to keep the font size fixed and use a monospace font, so I know exactly how long to truncate the URL to for any stamp size. The current implementation should work out of the box for any value of the stamp size, but you're right that it's not very tunable.

The opposite approach, which I think is what you're suggesting, is to make everything customizable. That's fine by me, but certain combinations of parameters will lead to horrible-looking stamps.

One alternative is to fix the font size relative to the size of the stamp. We don't have to do this programmatically -- we could just render it as it currently appears for the default stamp size, but then have ``stamp.size`` scale _everything_ up (including the text). Should be easy with a `\resizebox` after the TikZ graphic is rendered. That way we can ensure the stamp always looks good and keep the branding consistent across papers. What do you think?

Reply:
The reason I asked about this is because I tried making the stamp smaller, but it starts overlapping, since the text doesn't scale:
<img width="637" alt="Screen Shot 2022-09-15 at 12 13 23 PM" src="https://user-images.githubusercontent.com/7593028/190454720-6dbb2b27-5e35-4f7e-b57e-be5db2ad3718.png">
Maybe the `max_url_len` needs to scale differently due to the radius of the text being different at the top of characters vs bottom of characters?

But in any case, I think either configurable fontsize would help this, or your suggestion of a `\resizebox` would probably be even better! An overall `stamp.size` would be awesome and the best way.

Reply:
Resizebox version (actually scalebox) implemented.